docs/adr/0012_oci_format_compatibility.md

```diff
@@ -0,0 +1,381 @@
+# Future Format Considerations for OCM Component Versions in OCI Repositories
+
+* **Status**: proposed
+* **Deciders**: OCM Technical Steering Committee
+* **Date**: 2025.10.15
+
+**Technical Story**:
+
+Provide a way to discover, read and update component versions from OCI Artifacts that have a top level OCI Image Index manifest instead of the descriptor manifest.
```

Comment:
love this proposal!
great step towards making OCM more compatible with established cloud-native tooling.
these are the fine tunings which heavily improve developer experience.